diff --git a/plugin/autocmds.vim b/plugin/autocmds.vim
index 526d6e3..5a852ac 100644
--- a/plugin/autocmds.vim
+++ b/plugin/autocmds.vim
@@ -32,6 +32,10 @@ augroup END
 
 function! s:FmtIfAutoEnabled() abort
   if get(b:, 'codefmt_auto_format_buffer')
-    call codefmt#FormatBuffer()
+    if get(g:, 'codefmt_silent_auto_format')
+      silent! call codefmt#FormatBuffer()
+    else
+      call codefmt#FormatBuffer()
+    endif
   endif
 endfunction
