diff --git a/autoload/airline/extensions/ale.vim b/autoload/airline/extensions/ale.vim
index 4b19bca..4a8447f 100644
--- a/autoload/airline/extensions/ale.vim
+++ b/autoload/airline/extensions/ale.vim
@@ -4,7 +4,7 @@
 
 scriptencoding utf-8
 
-if !get(g:, 'loaded_ale_dont_use_this_in_other_plugins_please', 0)
+if !get(g:, 'loaded_ale_dont_use_this_in_other_plugins_please', 0) && !get(g:, 'airline_always_load_ale', 0)
   finish
 endif
 
